server/src/omtDocumentInformationProvider.ts

```diff
@@ -0,0 +1,359 @@
+                fileImportsResult && calledObjects.push(...getLocalLocationsForCode(fileImportsResult.definedObjects, l, line));
+            }
+        }
+        console.timeEnd('getOmtDocumentInformation for ' + document.uri);
+        return {
+            documentLinks,
+            definedObjects: fileImportsResult?.definedObjects ?? [],
+            calledObjects,
+            availableImports: fileImportsResult?.availableImports ?? []
+        };
+    }
+}
+
+function getUriMatch(line: string, document: TextDocument, shorthands: Map<string, string>) {
+    const uriMatch = omtUriMatch.exec(line);
+    if (uriMatch) {
+        // match[0] is the full match inluding the whitespace of match[1]
+        // match[1] is the whitespace and optional quotes. both of which we don't want to include in the linked text
+        // match[2] is the link text, including the @shorthands
+        const link = replaceStart(uriMatch[2].trim(), shorthands);
+        const url = isAbsolute(link) ? resolve(document.uri, link) : toAbsolutePath(document, link);
+        return { uriMatch, url };
+    }
+}
+
+function getDiMatch(line: string) {
+    const diMatch = declaredImportMatch.exec(line);
+    if (diMatch) {
+        // match[0] is the full line match including the trailing colon
+        // match[1] is the whitespace prepending the import
+        // match[2] is the module name
+        const declaredImportModule = '' + diMatch[2];
+        return { diMatch, declaredImportModule }
+    }
+}
+
+function getReferencesToOtherFilesForCode(fileImports: OmtImport[], l: number, line: string): OmtLocalObject[] {
+    return findUsagesInLine(fileImports.map(x => x.name), l, line);
+}
+
+function getLocalLocationsForCode(declaredObjects: OmtLocalObject[], l: number, line: string): OmtLocalObject[] {
+    return findUsagesInLine(declaredObjects.map(x => x.name), l, line);
+}
+
+function findUsagesInLine(declaredObjects: string[], l: number, line: string): OmtLocalObject[] {
+    const documentLinks: OmtLocalObject[] = [];
+
+    declaredObjects.filter(declaredObject => line.includes(declaredObject)).forEach(declaredObject => {
+        const characterIndex = line.search(new RegExp(`${declaredObject}(?=[^a-zA-Z0-9]|$)`)) ?? line.indexOf(declaredObject);
+
+        if (characterIndex >= 0) {
+            documentLinks.push({
+                name: declaredObject,
+                range: Range.create({ line: l, character: characterIndex }, { line: l, character: characterIndex + declaredObject.length })
+            });
+        }
+    });
+    return documentLinks;
+}
+
+export function getAvailableObjectsFromDocument(document: TextDocument, shorthands?: Map<string, string>): OmtAvailableObjects {
+    const documentText = document.getText();
+    const yamlDocument = parse(document.getText());
+    const availableImports: OmtImport[] = [];
+    const definedObjects: OmtLocalObject[] = [];
+    if ("import" in yamlDocument && shorthands) {
```

Comment:
We really should try to combine this reading of the imports with the other one further back in this file. That would save a lot of code. I think the use of the yaml parser here is better than just parsing the file as text

Reply:
I agree, I also prefer the yaml parser, when using that we don't have to use regular expressions anymore with three different groups (whitespace followed by quotes followed by text). It will make it a lot easier to understand.

- [ ] create document links using the YAML parser